src/service/GPUEnergyService6DModes.tpp

```diff
@@ -260,53 +275,33 @@ public:
 			/* Perform cuda kernel calls */
 			size_t gridSizeLig = ( numElLig + BLSZ_TRAFO - 1) / BLSZ_TRAFO;
 			size_t gridSizeRec = ( numElRec + BLSZ_TRAFO - 1) / BLSZ_TRAFO;
-			size_t gridSize = ( max(numElRec, numElLig) + BLSZ_TRAFO - 1) / BLSZ_TRAFO;
+
 
 			/* Device: Wait for completion of copyH2D of DOFs to complete */
 			cudaVerify(cudaStreamWaitEvent(streams[2], events[0], 0));
 
-//			d_DOF2Pos(
-//					BLSZ_TRAFO,
-//					gridSize,
-//					streams[2],
-//					stageResc.lig->xPos,
-//					stageResc.lig->yPos,
-//					stageResc.lig->zPos,
-//					d_dof[pipeIdx[1]].get(0),
-//					stageResc.lig->numAtoms,
-//					it->size(),
-//					d_trafoLig.getX(),
-//					d_trafoLig.getY(),
-//					d_trafoLig.getZ()); //OK
-
-			d_DOFPos(
+			 d_DOFPos<REAL, 0, true>(
 				BLSZ_TRAFO,
-				gridSize,
+				gridSizeRec,
 				streams[2],
-				stageResc.rec->xPos,
-				stageResc.rec->yPos,
-				stageResc.rec->zPos,
-				stageResc.lig->xPos,
-				stageResc.lig->yPos,
-				stageResc.lig->zPos,
-				stageResc.rec->xModes,
-				stageResc.rec->yModes,
-				stageResc.rec->zModes,
-				stageResc.lig->xModes,
-				stageResc.lig->yModes,
-				stageResc.lig->zModes,
+				stageResc.rec,
 				d_dof[pipeIdxDof[1]].get(0),
-				stageResc.rec->numAtoms,
-				stageResc.lig->numAtoms,
-				stageResc.rec->numModes,
-				stageResc.lig->numModes,
 				it->size(),
 				d_defoRec.getX(),
 				d_defoRec.getY(),
 				d_defoRec.getZ(),
 				d_trafoRec.getX(),
 				d_trafoRec.getY(),
-				d_trafoRec.getZ(),
+				d_trafoRec.getZ()
+				);
+
+			 d_DOFPos<REAL, 1, true>(
```

Comment:
We really need to talk about the strategy behind calling the DOFPos<>(and all the others) twice, instead of an optimized version of that (like transforming rec and lig at the same time). I think we lose a lot of perfomance since some work is repeated like reading the the DOFs from the memory!!

Reply:
I agree. For this I tried to follow your words, to make it simple to understand first and then optimize for performance later on. But in comparison to the old approach with both receptor and ligand being transformed at once, the workload per thread is more equally distributed. If the number of atoms of receptor and ligand differed, some threads hat to transform both and some just the protein with the higher number of atoms.